### ISSUE TITLE
Class weights.

### ISSUE THREAD

**Issue by CDC1688** [microsoft/LightGBM#2918]
Hey there,

I want to set loss function for imbalanced binary class. however, on the docs it says class weight only support "Use this parameter only for multi-class classification task; for binary classification task you may use is_unbalance or scale_pos_weight parameters. " .

I read from other post pos_weight parameters will change the distribution of my data, I can't change the distribution. How do I set the weight class for this situation? 

Thanks ,

**Comment by guolinke** [comment]
I think both class_weight and `scale_pos_weight`/`is_unbalance` will change the distribution. And there are the same when setting the corresponding values in class_weight.
